lib/rules/define-macros-order.js

```diff
@@ -0,0 +1,248 @@
+  if (!node.parent) {
+    throw new Error('Macros has parent')
+  }
+
+  if (node.parent.type === 'Program') {
+    return node
+  }
+
+  return getDefineMacrosStatement(node.parent)
+}
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+/** @param {RuleContext} context */
+function create(context) {
+  const scriptSetup = utils.getScriptSetupElement(context)
+
+  if (!scriptSetup) {
+    return {}
+  }
+
+  const sourceCode = context.getSourceCode()
+  const options = context.options
+  /** @type {[string, string]} */
+  const order = (options[0] && options[0].order) || DEFAULT_ORDER
+  /** @type {Map<string, ASTNode>} */
+  const macrosNodes = new Map()
+
+  return utils.compositingVisitors(
+    utils.defineScriptSetupVisitor(context, {
+      onDefinePropsExit(node) {
+        macrosNodes.set(MACROS_PROPS, getDefineMacrosStatement(node))
+      },
+      onDefineEmitsExit(node) {
+        macrosNodes.set(MACROS_EMITS, getDefineMacrosStatement(node))
+      }
+    }),
+    {
+      'Program:exit'(program) {
+        const shouldFirstNode = macrosNodes.get(order[0])
+        const shouldSecondNode = macrosNodes.get(order[1])
+        const firstStatementIndex = getStatementAfterImportsIndex(program)
+        const firstStatement = program.body[firstStatementIndex]
+
+        // have both defineEmits and defineProps
+        if (shouldFirstNode && shouldSecondNode) {
+          const secondStatement = program.body[firstStatementIndex + 1]
+
+          // need move only first
+          if (firstStatement === shouldSecondNode) {
+            reportNotOnTop(order[1], shouldFirstNode, firstStatement)
+            return
+          }
+
+          // need move both defineEmits and defineProps
+          if (firstStatement !== shouldFirstNode) {
+            reportBothNotOnTop(
+              shouldFirstNode,
+              shouldSecondNode,
+              firstStatement
+            )
+            return
+          }
+
+          // need move only second
+          if (secondStatement !== shouldSecondNode) {
+            reportNotOnTop(order[1], shouldSecondNode, shouldFirstNode)
+          }
+
+          return
+        }
+
+        // have only first and need to move it
+        if (shouldFirstNode && firstStatement !== shouldFirstNode) {
+          reportNotOnTop(order[0], shouldFirstNode, firstStatement)
+          return
+        }
+
+        // have only second and need to move it
+        if (shouldSecondNode && firstStatement !== shouldSecondNode) {
+          reportNotOnTop(order[1], shouldSecondNode, firstStatement)
+        }
+      }
+    }
+  )
+
+  /**
+   * @param {ASTNode} shouldFirstNode
+   * @param {ASTNode} shouldSecondNode
+   * @param {ASTNode} before
+   */
+  function reportBothNotOnTop(shouldFirstNode, shouldSecondNode, before) {
+    context.report({
+      node: shouldFirstNode,
```

Comment:
Here we show error only at `shouldFirstNode`, but move both `shouldFirstNode` and `shouldSecondNode`.
Because of this RuleTester behavior described here(https://github.com/eslint/eslint/issues/11187) I can't make two errors with two fixes(I tried and had the same problem)

Reply:
I think that's fine.